### PR TITLE
katex: Remove renderKaTex experimental flag

### PR DESCRIPTION
**Pull request by rajveermalviya**
This flag has been enabled since v0.0.29 (beta) release, and we don't plan to disable it. So remove the flag completely, simplifying related code.

Fixes: #1728

### CODE REVIEW

File: test/widgets/content_test.dart [hunk 1]
```diff
@@ -1017,50 +1002,27 @@ void main() {
...
-      const html = '<span class="katex">'
-        '<span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>λ</mi></mrow>'
-          '<annotation encoding="application/x-tex"> \\lambda </annotation></semantics></math></span>'
-        '<span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.6944em;"></span><span class="mord mathnormal">λ</span></span></span></span>';
-      await checkFontSizeRatio(tester,
-        targetHtml: html,
-        targetFontSizeFinder: mkTargetFontSizeFinderFromPattern(r'λ'));
-    }, skip: true // TODO(#46): adapt this test
-                  //   (it needs a more complex targetFontSizeFinder;
-                  //    see other uses in this file for examples.)
```
Comment (gnprice) [inline test/widgets/content_test.dart:1038]: nit:

    content test [nfc]: Drop duplicate skipped KaTeX font-size ratio test

rather, it's a skipped duplicate such test — there's another KaTeX font-size ratio test, which this is a duplicate of, but there's no other *skipped* KaTeX font-size ratio test, so it can't be a duplicate within that category.

File: test/widgets/content_test.dart [hunk 2]
```diff
@@ -1017,50 +1002,27 @@ void main() {
...
-      final globalSettings = testBinding.globalStore.settings;
-      await globalSettings.setBool(BoolGlobalSetting.renderKatex, false);
-
+    testWidgets('displays KaTeX content', (tester) async {
       await prepareContent(tester, plainContent(ContentExample.mathInline.html));
-      tester.widget(find.text(r'\lambda', findRichText: true));
+      tester.widget(find.text('λ', findRichText: true));
     });
 
-    testWidgets('displays KaTeX content; experimental flag enabled', (tester) async {
```
Comment (gnprice) [inline test/widgets/content_test.dart:1064]: nit: make summary line more specific

    content test [nfc]: Reorder KaTeX widget tests
    
    With renderKatex flag enabled (and soon to be default behaviour)
    being first.

There's room in that summary line to put some of that information from the body, to give an idea of what the reordering is about. Like:
> content test [nfc]: Reorder KaTeX widget tests, enabled before disabled

File: test/widgets/content_test.dart (hunk 1 above)
Comment (gnprice) [reply test/widgets/content_test.dart:1038]: Also please add a sentence to the commit message clarifying that the existing test this duplicates is the one above it, not the one below. I initially read it as the latter (because the descriptions match — my conflict-resolution error, probably, somewhere in the chain of cherry-picking that led to b073c6b26f785ff72530e23e768e58d33d30211b / #1270), and was going to comment that it's not actually a duplicate of that :slightly_smiling_face:

File: test/model/content_test.dart [hunk 3]
```diff
@@ -528,6 +527,20 @@ class ContentExample {
       ]),
     ]));
 
+  // A test message to test the fallback behaviour of KaTeX implementation.
+  static final mathInlineUnknown = ContentExample.inline(
+    'inline math',
+    r"$$ \lambda $$",
```
Comment (gnprice) [inline test/model/content_test.dart]: Since this HTML isn't something the server actually generates, this `markdown` field should be null. See its doc:
```dart
  /// The Zulip Markdown source, if any, that the server renders as [html].
  ///
  /// This is useful for reproducing the example content for live use in the
  /// app, and as a starting point for variations on it.
  ///
  /// Currently the test suite does not verify the relationship between
  /// [markdown] and [html].
  ///
  /// If there is no known Markdown that a Zulip server can render as [html],
  /// then this should be null and a comment should explain why the test uses
  /// such an example.
  final String? markdown;
```

So e.g., following the example `emojiUnicodeClassesFlipped`:

```suggestion
  static final mathInlineUnknown = ContentExample.inline(
    'inline math',
    null, // r"$$ \lambda $$" (hypothetical server variation)
```

File: test/widgets/content_test.dart [hunk 4]
```diff
@@ -1017,50 +1002,27 @@ void main() {
...
-                  //    see other uses in this file for examples.)
-    );
-
-    testWidgets('maintains font-size ratio with surrounding text, when showing TeX source', (tester) async {
-      addTearDown(testBinding.reset);
-      final globalSettings = testBinding.globalStore.settings;
-      await globalSettings.setBool(BoolGlobalSetting.renderKatex, false);
-
-      const html = '<span class="katex">'
+    testWidgets('maintains font-size ratio with surrounding text, when fallback to TeX source', (tester) async {
```
Comment (gnprice) [inline test/widgets/content_test.dart]: nit: "fallback" is a noun, "fall back" the verb:
```suggestion
    testWidgets('maintains font-size ratio with surrounding text, when falling back to TeX source', (tester) async {
```

**Review by gnprice — COMMENTED** [review]
Thanks for cleaning this up! Generally all looks good; small comments below.

**Comment by rajveermalviya** [comment]
Thanks for the review @gnprice! Pushed an update, PTAL.

File: test/model/content_test.dart [hunk 5]
```diff
@@ -511,7 +510,7 @@ class ContentExample {
 
   static final mathInline = ContentExample.inline(
     'inline math',
-    r"$$ \lambda $$",
+    null, // r"$$ \lambda $$" (hypothetical server variation)
```
Comment (gnprice) [inline test/model/content_test.dart]: I think you made this edit on a different example than intended :slightly_smiling_face:

Comment (rajveermalviya) [reply test/model/content_test.dart]: Ahh, should be fixed now.

**Comment by rajveermalviya** [comment]
Thanks for the review @gnprice! Pushed an update, PTAL.

**Comment by gnprice** [comment]
Thanks! Merging, with a tweak in one commit message (cf https://github.com/zulip/zulip-flutter/pull/1742#discussion_r2229237016):
```
$ git range-diff origin pr/1742 @
1:  edfb68198 ! 1:  8c9a7b0ce content test [nfc]: Drop skipped duplicate KaTeX font-size ratio test
    @@ Commit message
         This skipped test is a duplicate of the one above it, but
         incorrectly has the description of the one below it. Probably a
         result of conflict-resolution error in b073c6b26 which introduced
    -    this duplicate.
    +    this duplicate, or another in the chain of cherry-picks that led to
    +    that commit.
     
      ## test/widgets/content_test.dart ##
     @@ test/widgets/content_test.dart: void main() {
2:  95c25c6c3 = 2:  6459af689 content test [nfc]: Reorder KaTeX widget tests, enabled before disabled
3:  ccf262f4c = 3:  da5c1f706 katex: Remove renderKaTex experimental flag
```
